com.unity.render-pipelines.high-definition/Editor/Material/ShaderGraph/HDSubTarget.cs

```diff
@@ -66,13 +67,22 @@ public SystemData systemData
 
         public virtual string identifier => GetType().Name;
 
-        public virtual ScriptableObject GetMetadataObject()
+        public virtual ScriptableObject GetMetadataObject(GraphData graph)
         {
             var hdMetadata = ScriptableObject.CreateInstance<HDMetadata>();
             hdMetadata.shaderID = shaderID;
             hdMetadata.subTargetGuid = subTargetGuid;
             hdMetadata.migrateFromOldCrossPipelineSG = m_MigrateFromOldCrossPipelineSG;
             hdMetadata.hdSubTargetVersion = systemData.version;
+            hdMetadata.hasVertexModificationInMotionVector = systemData.customVelocity || systemData.tessellation;
```

Comment:
can you elaborate about the "|| systemData.tessellation" part? Why tessellation require motion vector?

Reply:
`hasVertexModificationInMotionVector` indicates if we have a vertex alteration during motion vector pass, so either modifying the vertex itself, adding a custom velocity or process a tessellation.

The tessellation is using a special path in [MotionVectorVertexShaderCommon](https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/MotionVectorVertexShaderCommon.hlsl#L210) where we are returning world space position instead of clip space position (the clip transform is applied later)

In that case, we can't use the precomputed projected clip position from VFX since it can be change afterwards.

However, the tessellation won't work with VFX right now, we could remove this condition.